src/pybind/mgr/dashboard/controllers/smb.py

```diff
@@ -88,6 +89,26 @@
 }
 
 
+def raise_on_failure(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        result = func(*args, **kwargs)
+
+        if 'success' in result and result['success'] is False:
+            msg = 'Operation was unsuccessful'
+
+            # Extracts the result msg from either of two possible response structures:
+            if 'results' in result:
+                msg = result['results'][0]['msg'] if 'msg' in result['results'][0] else msg
+            else:
+                msg = result['msg'] if 'msg' in result else msg
```

Comment:
How about this ? 
```suggestion
       if result.get('success') is False:
             msg = 'Operation was unsuccessful'
            # Extracts the result msg from either of two possible response structures:
            if 'results' in result:
                msg = result['results'][0].get('msg', msg) 
            elif 'msg' in result:
                msg = result['msg']          
```

Reply:
Thanks, that makes it more readable. Updated it